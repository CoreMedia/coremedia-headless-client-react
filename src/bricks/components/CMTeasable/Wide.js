import React from "react";
import styled from "styled-components";

import Picture from "../CMPicture";
import CTA from "../CTA";
import OptionalLink from "../Common/OptionalLink";
import Strings from "../../../l18n";
import getColors from "../../../styles/colors";
import {H2, P, TeaserBox} from "../../../styles";

const PictureBox = styled.div`
  width: 100%;
  @media only screen and (min-width: 48em) {
    width: 50%;
  }
`;
PictureBox.displayName = 'WideTeaserPictureBox';

const TextBox = styled.div`
  color: ${props => props.colors.fg};
  background-color: ${props => props.colors.bgsolid};
  text-align: center;
  @media only screen and (min-width: 48em) {
    position: absolute;
    width: 50%;
    height: 100%;
    left: 50%;
    top: 0%;
  }
`;
TextBox.displayName = 'WideTeaserTextBox';

const TextBoxWrapper = styled.div`
  padding: 2rem;
  @media only screen and (min-width: 48em) {
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
  }
`;
TextBoxWrapper.displayName = 'WideTeaserTextBoxWrapper';

const Headline = H2.extend`
  padding: 0;
`;
Headline.displayName = 'WideTeaserHeadline';

const Text = P.extend`
  margin-top: 1em;
  padding: 0;
`;
Text.displayName = 'WideTeaserText';

const composeContent = (url, pictureLink: string, pictureTitle: string, pictureAlt: string, title?: string, text?: string, params?: any) => {
  const colors = getColors(params.color);
  const content = [];
  if (pictureLink) {
    content.push(
      <PictureBox key="picturebox">
        <Picture key="picture"
                 link={pictureLink}
                 ratio="landscape_ratio3x2"
                 title={pictureTitle}
                 alt={pictureAlt}
                 colors={colors}
        />
      </PictureBox>
    );
  }
  if (title || text || params.ctaShow) {
    content.push(
      <TextBox key="textbox" colors={colors}>
        <TextBoxWrapper>
          {title && <Headline>{title}</Headline>}
          {text && <Text dangerouslySetInnerHTML={{__html: text}}/>}
          {params.ctaShow && !!url && (<CTA text={params.ctaText ? params.ctaText : Strings.teaser_cta_text}/>)}
        </TextBoxWrapper>
      </TextBox>
    );
  }
  return content;
};

type Props = {
  url?: string,
  pictureLink?: string,
  pictureTitle?: string,
  pictureAlt?: string,
  title?: string,
  text?: string,
  params?: any
};

const WideTeaser = ({url, pictureLink, pictureTitle, pictureAlt, title, text, params = {},}: Props) => {
  return (
    <OptionalLink url={url}>
      <TeaserBox>
        {composeContent(url, pictureLink, pictureTitle, pictureAlt, title, text, params)}
      </TeaserBox>
    </OptionalLink>
  );
};

export default WideTeaser;
