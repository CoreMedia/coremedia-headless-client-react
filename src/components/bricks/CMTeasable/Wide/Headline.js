// @flow
import { H2 } from '../../../basic/Heading';

const Headline = H2.extend`
  padding: 0;
`;

Headline.displayName = 'Headline';

Headline.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
    typography: {
      fontFamily: {
        heading: 'Helvetica',
      },
      fontWeight: {
        normal: 400,
      },
    },
  },
};

export default Headline;
