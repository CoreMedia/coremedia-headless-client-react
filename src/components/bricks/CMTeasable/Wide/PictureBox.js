// @flow
import styled from 'styled-components';
import { media } from '../../../../styles/themes/utils';

const PictureBox = styled.div`
  width: 100%;
  ${props => media(props.theme.breakpoints.tablet)`width: 50%;`};
`;

PictureBox.displayName = 'PictureBox';

PictureBox.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
  },
};

export default PictureBox;
