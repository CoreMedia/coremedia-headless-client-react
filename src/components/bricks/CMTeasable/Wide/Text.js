// @flow
import { P } from '../../../basic/Paragraph';

const Text = P.extend`
  margin-top: 1em;
  padding: 0;
`;

Text.displayName = 'Text';

Text.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
    name: 'default',
  },
};

export default Text;
