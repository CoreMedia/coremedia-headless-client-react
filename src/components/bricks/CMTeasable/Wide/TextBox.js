// @flow
import styled from 'styled-components';
import { media, getColors } from '../../../../styles/themes/utils';

const TextBox = styled.div`
  color: ${props => getColors(props.theme.colors, props.color).fg};
  background-color: ${props => getColors(props.theme.colors, props.color).bgsolid};
  text-align: center;
  ${props => media(props.theme.breakpoints.tablet)`
    position: absolute;
    width: 50%;
    height: 100%;
    left: 50%;
    top: 0%;
  `};
`;

TextBox.displayName = 'TextBox';

TextBox.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
    colors: {
      default: {
        fg: '#fafafa',
        bgsolid: 'rgba(239, 223, 15, 1)',
        bgtrans: 'rgba(239, 223, 15, 0.7)',
        huerotation: '5deg',
      },
    },
  },
};

export default TextBox;
