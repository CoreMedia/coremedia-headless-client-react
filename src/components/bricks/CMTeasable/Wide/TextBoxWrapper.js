// @flow
import styled from 'styled-components';
import { media } from '../../../../styles/themes/utils';

const TextBoxWrapper = styled.div`
  padding: 2rem;
  ${props => media(props.theme.breakpoints.tablet)`
    position: absolute;
    top: 50%;
    transform: translateY(-50%);
  `};
`;

TextBoxWrapper.displayName = 'TextBoxWrapper';

TextBoxWrapper.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
    },
  },
};

export default TextBoxWrapper;
