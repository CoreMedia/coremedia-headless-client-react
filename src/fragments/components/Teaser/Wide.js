import React from "react";

import {Wide as TeaserBrick} from "../../../bricks/components/CMTeasable";

const WideTeaser = {
  type: 'teasables',
  view: null,
  brick(data, params) {
    const picture = data.picture || {};
    return (
      <TeaserBrick url={params.url ? params.url : data.teaserTarget.link}
                   pictureLink={picture.link}
                   pictureTitle={picture.title}
                   pictureAlt={picture.alt}
                   title={data.teaserTitle}
                   text={data.teaserText}
                   params={params}
      />
    );
  }
};

export default WideTeaser;
