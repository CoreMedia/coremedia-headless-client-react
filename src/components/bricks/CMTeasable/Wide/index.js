// @flow
import * as React from 'react';
import PropTypes from 'prop-types';

import Headline from './Headline';
import PictureBox from './PictureBox';
import Text from './Text';
import TextBox from './TextBox';
import TextBoxWrapper from './TextBoxWrapper';
import Picture from '../../CMPicture';
import CTA from '../../CTA';
import { TeaserBox } from '../../../basic/Box';
import { OptionalLink } from '../../../basic/Link';
import Strings from '../../../../l18n';
import type { Teaser } from '../../../../types';

const composeContent = ({
  url,
  pictureLink,
  pictureTitle,
  pictureAlt,
  title,
  text,
  params,
}: Teaser): React.Node => {
  const color = params && params.color;
  const content = [];
  if (pictureLink) {
    content.push(
      <PictureBox key="picturebox">
        <Picture
          key="picture"
          link={pictureLink}
          ratio="landscape_ratio3x2"
          title={pictureTitle}
          alt={pictureAlt}
          color={color}
        />
      </PictureBox>
    );
  }
  if (title || text || (params && params.ctaShow)) {
    content.push(
      <TextBox key="textbox" color={color}>
        <TextBoxWrapper>
          {title && <Headline>{title}</Headline>}
          {text && <Text dangerouslySetInnerHTML={{ __html: text }} />}
          {params &&
            params.ctaShow &&
            !!url && (
              <CTA
                text={params && params.ctaText ? params && params.ctaText : Strings.teaser_cta_text}
              />
            )}
        </TextBoxWrapper>
      </TextBox>
    );
  }
  return content;
};

const WideTeaser = ({
  url,
  pictureLink,
  pictureTitle,
  pictureAlt,
  title,
  text,
  params = {},
}: Teaser) => {
  return (
    <OptionalLink url={url}>
      <TeaserBox>
        {composeContent({ url, pictureLink, pictureTitle, pictureAlt, title, text, params })}
      </TeaserBox>
    </OptionalLink>
  );
};

WideTeaser.propTypes = {
  url: PropTypes.string,
  pictureLink: PropTypes.string,
  pictureTitle: PropTypes.string,
  pictureAlt: PropTypes.string,
  title: PropTypes.string,
  text: PropTypes.string,
  params: PropTypes.shape({
    color: PropTypes.string,
    ctaShow: PropTypes.bool,
    ctaText: PropTypes.string,
    url: PropTypes.string,
  }),
};

export default WideTeaser;
