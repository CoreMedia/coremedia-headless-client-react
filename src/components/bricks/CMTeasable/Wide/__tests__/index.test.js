// @flow
import React from 'react';

import Wide from '..';

describe('WideTeaser Component', () => {
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        text="Discover the new CoreMedia Headless Services."
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        title="Content Management at a new Scale with CoreMedia CaaS"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link, Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link and Picture', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link and CTA', () => {
    const wrapper = shallow(
      <Wide params={{ color: 'blue', ctaShow: true, url: '/caas.html' }} url="/caas.html" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link and CTA (custom Label)', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', ctaShow: true, ctaText: 'Label', url: '/caas.html' }}
        url="/caas.html"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Link', () => {
    const wrapper = shallow(
      <Wide params={{ color: 'blue', url: '/caas.html' }} url="/caas.html" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser without CTA', () => {
    const wrapper = shallow(<Wide params={{ color: 'blue', ctaShow: true, url: '/caas.html' }} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Picture', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Picture (without params)', () => {
    const wrapper = shallow(
      <Wide pictureAlt="Globe" pictureLink="coremedia:///image/2656/data" pictureTitle="Globe" />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render WideTeaser with Picture and CTA', () => {
    const wrapper = shallow(
      <Wide
        params={{ color: 'blue', ctaShow: true, url: '/caas.html' }}
        pictureAlt="Globe"
        pictureLink="coremedia:///image/2656/data"
        pictureTitle="Globe"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
