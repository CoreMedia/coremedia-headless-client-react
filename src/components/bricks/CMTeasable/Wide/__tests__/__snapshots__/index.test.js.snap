// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WideTeaser Component should render WideTeaser with Link 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox />
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link and CTA (custom Label) 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <CTA
          text="Label"
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link and CTA 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <CTA
          text="More"
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link and Picture 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 1`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <Headline
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "heading": "Helvetica",
                },
                "fontWeight": Object {
                  "normal": 400,
                },
              },
            }
          }
        >
          Content Management at a new Scale with CoreMedia CaaS
        </Headline>
        <Text
          dangerouslySetInnerHTML={
            Object {
              "__html": "Discover the new CoreMedia Headless Services.",
            }
          }
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "text": "Helvetica",
                },
              },
            }
          }
        />
        <CTA
          text="More"
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 2`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <Headline
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "heading": "Helvetica",
                },
                "fontWeight": Object {
                  "normal": 400,
                },
              },
            }
          }
        >
          Content Management at a new Scale with CoreMedia CaaS
        </Headline>
        <Text
          dangerouslySetInnerHTML={
            Object {
              "__html": "Discover the new CoreMedia Headless Services.",
            }
          }
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "text": "Helvetica",
                },
              },
            }
          }
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 3`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <Headline
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "heading": "Helvetica",
                },
                "fontWeight": Object {
                  "normal": 400,
                },
              },
            }
          }
        >
          Content Management at a new Scale with CoreMedia CaaS
        </Headline>
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 4`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <Text
          dangerouslySetInnerHTML={
            Object {
              "__html": "Discover the new CoreMedia Headless Services.",
            }
          }
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "text": "Helvetica",
                },
              },
            }
          }
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 5`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <Text
          dangerouslySetInnerHTML={
            Object {
              "__html": "Discover the new CoreMedia Headless Services.",
            }
          }
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "text": "Helvetica",
                },
              },
            }
          }
        />
        <CTA
          text="More"
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 6`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <Headline
          theme={
            Object {
              "breakpoints": Object {
                "desktop": 992,
                "large": 1280,
                "tablet": 768,
              },
              "typography": Object {
                "fontFamily": Object {
                  "heading": "Helvetica",
                },
                "fontWeight": Object {
                  "normal": 400,
                },
              },
            }
          }
        >
          Content Management at a new Scale with CoreMedia CaaS
        </Headline>
        <CTA
          text="More"
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Link, Picture and CTA 7`] = `
<OptionalLink
  url="/caas.html"
>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      >
        <CTA
          text="More"
        />
      </TextBoxWrapper>
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Picture (without params) 1`] = `
<OptionalLink>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Picture 1`] = `
<OptionalLink>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser with Picture and CTA 1`] = `
<OptionalLink>
  <TeaserBox>
    <PictureBox
      key="picturebox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
        }
      }
    >
      <WithTheme(Picture)
        alt="Globe"
        color="blue"
        key="picture"
        link="coremedia:///image/2656/data"
        ratio="landscape_ratio3x2"
        title="Globe"
      />
    </PictureBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      />
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;

exports[`WideTeaser Component should render WideTeaser without CTA 1`] = `
<OptionalLink>
  <TeaserBox>
    <TextBox
      color="blue"
      key="textbox"
      theme={
        Object {
          "breakpoints": Object {
            "tablet": 768,
          },
          "colors": Object {
            "default": Object {
              "bgsolid": "rgba(239, 223, 15, 1)",
              "bgtrans": "rgba(239, 223, 15, 0.7)",
              "fg": "#fafafa",
              "huerotation": "5deg",
            },
          },
        }
      }
    >
      <TextBoxWrapper
        theme={
          Object {
            "breakpoints": Object {
              "tablet": 768,
            },
          }
        }
      />
    </TextBox>
  </TeaserBox>
</OptionalLink>
`;
